Makefile

```diff
@@ -28,7 +43,30 @@ publish: $(JS)
 cli:
 	cd cli && python setup.py sdist bdist_wheel
 
+doc: $(VIRTUAL_ENV)/lib/site-packages/sphinx_js/__init__.py
+	$(MAKE) -C docs clean html
+
 clean:
-	rm -f $(JS)
+	rm -rf $(JS) venv .npm_installed
+
+
+# Private targets:
+
+# Make a virtualenv at $VIRTUAL_ENV if there isn't one or if requirements have
+# changed. Install the dev requirements and the actual requirements.
+$(VIRTUAL_ENV)/pyvenv.cfg: tooling/dev-requirements.txt cli/setup.py
+	$(PYTHON3) -m venv $(VIRTUAL_ENV)
+	pip3 install -r tooling/dev-requirements.txt
+	pip3 install -e cli
```

Comment:
Windows + pip requires: `pip3 install -e cli -f https://download.pytorch.org/whl/torch_stable.html` The addition of `dependency_links` only covers the `python setup.py install` case.

Reply:
I'm afraid that really is the case. :-( They seem to have removed both the `--process-dependency-links` and the `--allow-all-external` options, making this repetition necessary.

Reply:
What's more, https://www.python.org/dev/peps/pep-0508/, which is the closest thing to a replacement, doesn't support the flexibilty we need in its URL-based requirements: it provides only the ability to link directly to a specific file. We don't want to pin down the dependency to that degree. So, old cruft it is!